File: SchoolAngular/src/app/service/teacher.subject.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {TeacherSubject} from "../model/teacher.subject.model";

@Injectable()
export class TeacherSubjectService {

  private basUrl = "http://localhost:8080/teachersubject"

  constructor(private httpClient: HttpClient) {

  }

  getTeacherSubjectList(): Observable<TeacherSubject[]> {
    return this.httpClient.get<TeacherSubject[]>(`${this.basUrl}`);
  }

  findTeacherSubjectListForTeacher(teacherId: number): Observable<TeacherSubject[]> {
    return this.httpClient.get<TeacherSubject[]>(`${this.basUrl}/find-for-teacher/${teacherId}`);
  }

  createTeacherSubject(teacherSubject: TeacherSubject): Observable<TeacherSubject> {
    return this.httpClient.post<TeacherSubject>(`${this.basUrl}`, teacherSubject);
  }

  getTeacherSubjectById(id: number): Observable<TeacherSubject>{
    return this.httpClient.get<TeacherSubject>(`${this.basUrl}/${id}`);
  }

  updateTeacherSubject(id:number, teacherSubject:TeacherSubject): Observable<Object>{
    return this.httpClient.put(`${this.basUrl}/${id}`, teacherSubject);
  }

  deleteTeacherSubject(id:number): Observable<Object>{
    return this.httpClient.delete(`${this.basUrl}/${id}`);
  }

}

File: springCurseWork/src/main/java/com/example/springcursework/servise/TeacherSubjectService.java
package com.example.springcursework.servise;

import com.example.springcursework.model.TeacherSubject;

import java.util.List;

public interface TeacherSubjectService {
    public TeacherSubject insert(TeacherSubject teacherSubjectVO);
    public List<TeacherSubject> findAll();
    public void delete(int id);
    public TeacherSubject findById(int id);
    public TeacherSubject updateTeacherSubject(int id, TeacherSubject teacherSubjectVO);
    public List<TeacherSubject> findTeacherSubjectListForTeacher(int teacherId);
}
